src/application_config/app_config.h

```diff
@@ -319,28 +382,31 @@
 
 // ***** Flash storage constants *****/
 
-#define APP_FLASH_PAGES (16U) //!< 64 kB flash storage if page size is 4 kB.
-#define APP_FLASH_LOG_DATA_RECORDS_NUM   (APP_FLASH_PAGES - 2U) //!< swap page + settings.
+/* @ brief Caculate _RECORDS_NUM (actually blocks of log elements)  */
+// // // // // // // // // // // // // // // // // // // //
+//  From .ld:      stop_storage_flash =.+0x15000
+#define APP_FLASH_PAGES                 (0x15000/4096)-1             //!<  page size is 4 kB
```

Comment:
While the linker has some extra space, current specification is "10 days of logged data". 0x10 pages is enough for that, there is no reason to reserve extra 0x5 pages for application flash. The space might be needed for something else in the future.

Reply:
1)  Fourteen days of space is a much better "stated" limited. There are remote sensors (after all that's what the log is all about) that are visited weekly. Occasionally a visit cannot be made and the next visit should be able to retrieve the missed data. When did you ever hear of a user complaining that he got more space then he expected?

2) 16 pages of flash is may not be  enough if the interval between log entries is reduced.

3) Why artificially restrict the space. Nordic went to a lot of trouble to make flash big. There is no reason to restrict (waste) it. I would even advocate for additional pages  for ruuvitag_b since there is available flash space in that configuration. The last address used by code in the debug variant (including integration test) is currently at 4C750. Dropping the low address of FDS to 4D000 would allow for many more pages while still allowing code expansion of over 2,000 bytes. Basing the calculation of app_flash_pages on the reserved space size makes it clear how it is calculated (no magic number for mugles like me.) Choosing a more conservative value like 4E000 would anticipate a 6KB increase in code.

Reply:
There is always going to be someone who relies on current features. While it would be attractive to use every last byte of space available, it would cause issues in the future when e.g. sensor needs to be added to the firmware. 

At that point if there is no unused space, we need to trim the logging space and downgrade features or start supporting more variants. Neither is a good outcome for the most of users who want one stable "good enough" product